### PR TITLE
Fix financials formatting

### PR DESCRIPTION
**Pull request by ValueRaider**
Add whitespace to financials tables keys, e.g. "FreeCashFlow" -> "Free Cash Flow" #1186

Also restored @git-shogg code for creating nested index but disabled, eventually will be good to complete.

### CODE REVIEW

**Comment by git-shogg** [comment]
@ValueRaider great job! I have another idea on how we might include the parent->child index. Will look to raise a PR at a later date.

**Comment by fredrik-corneliusson** [comment]
I do not favor space separation for columns and or row names. 
The reason is as it makes working with yfinance in an interactive environment  (jupyter, ipython etc.) inconvenient as breaks double click to select column names for example.

**Comment by ValueRaider** [comment]
This just restores original yfinance behaviour. I think some spacing is necessary for readability, but can switch to '_'.

**Comment by fredrik-corneliusson** [comment]
Sounds like solution. 
If that is done would it be better do do the "presentation" formatting higher up in the call chain (the get_* methods in Ticker and base)? 
https://github.com/ranaroussi/yfinance/blob/ab1476c0d17d6cbc5c773541436f86b8fa917a35/yfinance/base.py#L760
That way the code internally does not need to deal with prettified indexes and column values, reducing code complexity and risks for bugs.
As a added bonus users of the API could then decide what format they want by adding a optional "raw" keyword argument or something to the API function.

**Comment by ValueRaider** [comment]
Done.

Question - should the Ticker properties (e.g. `Ticker.income_stmt`) default to use this formatting? README list these as way to access financials rather than `get_*()`, but user can't pass arguments through property.

**Review by fredrik-corneliusson — CHANGES_REQUESTED** [review]
Nice with tests 👍 
Just some code cleanups and escaping of variables used in dynamically created regexps needed.

File: yfinance/utils.py [hunk 1]
```diff
@@ -241,8 +241,34 @@ def format_quarterly_financial_statement(_statement, level_detail, order):
...
+        raise Exception("camel2title() 'sep' argument must be single character not:", sep)
+
+    if acronyms is None:
+        return [_re.sub("([a-z])([A-Z])", r"\g<1>{}\g<2>".format(sep), i).title() for i in o]
+
+    # Handling acronyms requires more care. Assumes Yahoo returns acronym strings upper-case
+
+    # Insert 'sep' between lower-then-upper-case
+    pat = "([a-z])([A-Z])"
+    rep = r"\g<1>{}\g<2>".format(sep)
```
[the remark this answers is not in the document]
Comment (ValueRaider) [reply yfinance/utils.py:255]: I'll resolve by not allowing special characters.

File: yfinance/utils.py [hunk 2]
```diff
@@ -241,8 +241,34 @@ def format_quarterly_financial_statement(_statement, level_detail, order):
...
+    o = [_re.sub(pat, rep, i) for i in o]
+
+    # Insert 'sep' after acronyms. Assumes Yahoo returns acronym strings upper-case
+    if not isinstance(acronyms, (set,list)):
+        raise Exception("camel2title() 'acronyms' argument should be an iterable of acronym strings")
+    for a in acronyms:
+        pat = "("+a+")" + "([A-Z][a-z])"
+        rep = r"\g<1>{}\g<2>".format(sep)
+        o = [_re.sub(pat, rep, i) for i in o]
```
[the remark this answers is not in the document]
Comment (ValueRaider) [reply yfinance/utils.py]: What practical difference does this make?

[the remark this answers is not in the document]
Comment (fredrik-corneliusson) [reply yfinance/utils.py]: Mainly it is the "o" name that makes the code hard to understand. it should be renamed first. It is for readability and maintainability. Also if input variable is mutable also O and 0 looks similar and is especially bad as a variabel name.
https://dev.to/alexomeyer/10-must-know-patterns-for-writing-clean-code-with-python-56bf

[the remark this answers is not in the document]
Comment (ValueRaider) [reply yfinance/utils.py]: I agree with descriptive names, just didn't understand why it has to differ from input when it isn't mutated.

[the remark this answers is not in the document]
Comment (fredrik-corneliusson) [reply yfinance/utils.py]: It's really not that important, it all depends on the descriptive name the variable will get, I figure.

**Pull request merged**